app/components/viral/base_component.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Viral
+  # Base component to be inherited from
+  class BaseComponent < Viral::Component
+    def initialize(tag:, classes: nil, **system_arguments)
```

Comment:
This comment applies to all the components in which you access the variable from the initialize method but will just leave the comment in this one place as to not clutter up the PR

When needing access to an instance variable we should be using `attr_accessor` to provide access to it from the rest of the class.

So here above the initialize method you would have:

`attr_accessor :tag, :content_tag_args`

Then when needing access to them in the class you would just use `tag` and `content_tag_args`. When updating this, the initialize methods do not need to be changed at all, and if the class doesn't have anything other than an initialize method then we don't need to add the attr_accessor

There's a few reasons for doing it this way. First it helps with some self documenting of the code, provides better error messages if a typo happens in the class with the variable name, and last but not least just for consistency across the application. We've been using attr_accessor in our services rather than accessing the instance variable directly from the initialize method

Reply:
Cleaned up in 9ea555d